src/routes/safe/container/index.tsx

```diff
@@ -32,7 +32,7 @@ const Container = (): React.ReactElement => {
   const featuresEnabled = useSelector(currentSafeFeaturesEnabled)
   const { address, owners } = useSelector(currentSafe)
   const addressFromUrl = extractSafeAddress()
-  const safeAddress = address || addressFromUrl
+  const safeAddress = addressFromUrl || address
```

Comment:
There was a race condition because the address from the store "lags" behind and causes an additional fetch call with the old address when switching safes.